lib/SILGen/SILGenProlog.cpp

```diff
@@ -449,6 +449,24 @@ void SILGenFunction::emitProlog(CaptureInfo captureInfo,
       }
     }
   }
+  
+  if (auto *funcDecl = dyn_cast_or_null<AbstractFunctionDecl>(FunctionDC->getAsDecl())) {
```

Comment:
If you're going to keep doing this in SILGen, how about passing this down as a parameter to emitProlog()? The caller knows the function being emitted, and I'd like to remove FunctionDC eventually since it's used for various hacks that should go away.

Reply:
Can I leave this up to you? emitProlog is called from multiple places (which I'm not familiar with) and I don't want to introduce a bug with this refactoring.